interface/javascript/main.js

```diff
@@ -750,3 +762,43 @@ function updateMap(mymap) {
                 return coords.map(i => i.x + ',' + i.y).join(' ');
             });
 }
+
+function openWeatherMapo(mymap, layer) {
+  return L.tileLayer('https://tile.openweathermap.org/map/{layer}/{z}/{x}/{y}.png?appid={api_key}', {
+   layer: layer,
+   maxZoom:17,
+   api_key: '2733c9a9c041a4ba7ce1963ae1a97dd4'
+   })
+}
+
+function heatMapMaker(mymap, cacheData, mode) {
+  // console.log(cacheData[0])
+
+  if (mode === "label") {
+    var heatCoordinates = cacheData.map(x => [x.feature.geometry.coordinates[0][0][0][1],x.feature.geometry.coordinates[0][0][0][0], x.label])
+  } else if (mode === "prediction") {
+    var heatCoordinates = cacheData.map(x => [x.feature.geometry.coordinates[0][0][0][1],x.feature.geometry.coordinates[0][0][0][0], x.prediction])
```

Comment:
Would it be possible to add a legend, or some kind short description? From the interface it is not totally clear what quantities are displayed and what the colours mean.

Reply:
yes, a legend will be helpful... we're still developing this and this may change soon... will keep this in mind for the next commit! Also, the colours were chosen randomly for distinction, we need to pick them more intuitively for good UX...